lib/commands/publish.js

```diff
@@ -131,6 +131,13 @@ class Publish extends BaseCommand {
 
     const resolved = npa.resolve(manifest.name, manifest.version)
 
+    const latestDistTag = await this.#latestPublishedVersion(resolved)
+    const latestTagIsGreater = latestDistTag ? semver.gte(latestDistTag, manifest.version) : false
+
+    if (latestTagIsGreater && isDefaultTag) {
+      throw new Error('Cannot publish a lower version without an explicit dist tag.')
```

Comment:
nit, if you like a bit of whimsy:
```suggestion
      throw new RangeError('Cannot publish a lower version without an explicit dist tag.')
```